File: SlimFileManager/src/main/java/com/slim/slimfilemanager/settings/SettingsProvider.java
/*
 * Copyright (C) 2015 The SlimRoms Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.slim.slimfilemanager.settings;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.text.TextUtils;

import com.slim.slimfilemanager.widget.TabItem;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class SettingsProvider {

    // File Manager
    public static final String KEY_ENABLE_ROOT = "enable_root";
    public static final String THEME = "app_theme";
    public static final String SORT_MODE = "sort_mode";
    public static final String SMALL_INDICATOR = "small_indicator";

    // Text Editor
    public static final String USE_MONOSPACE = "use_monospace";
    public static final String EDITOR_WRAP_CONTENT = "editor_wrap_content";
    public static final String SUGGESTION_ACTIVE = "suggestion_active";
    public static final String EDITOR_ENCODING = "editor_encoding";
    public static final String FONT_SIZE = "font_size";
    public static final String SPLIT_TEXT = "page_system_active";

    public static SharedPreferences get(Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }

    public static SharedPreferences.Editor put(Context context) {
        return get(context).edit();
    }

    public static void put(Context context, String key, Object o) {
        if (o instanceof Boolean) {
            put(context).putBoolean(key, (Boolean) o).apply();
        } else if (o instanceof String) {
            put(context).putString(key, (String) o).apply();
        } else if (o instanceof Integer) {
            put(context).putInt(key, (Integer) o).apply();
        }
    }

    public static String getString(Context context, String key, String defValue) {
        return get(context).getString(key, defValue);
    }

    public static void putString(Context context, String key, String value) {
        put(context).putString(key, value).apply();
    }

    public static boolean getBoolean(Context context, String key, boolean defValue) {
        return get(context).getBoolean(key, defValue);
    }

    public static void putBoolean(Context context, String key, boolean b) {
        put(context).putBoolean(key, b).apply();
    }

    public static int getInt(Context context, String key, int defValue) {
        int i;
        try {
            i = get(context).getInt(key, defValue);
        } catch (Exception e) {
            i = Integer.parseInt(get(context).getString(key, Integer.toString(defValue)));
        }
        return i;
    }

    public static void putInt(Context context, String key, int value) {
        put(context).putInt(key, value).commit();
    }

    public static ArrayList<TabItem> getTabList(Context context, String key,
                                                ArrayList<TabItem> def) {
        ArrayList<TabItem> items = new ArrayList<>();
        ArrayList<String> array = new ArrayList<>(
                Arrays.asList(TextUtils.split(get(context).getString(key, ""), "‚‗‚")));
        for (String s : array) {
            items.add(TabItem.fromString(s));
        }
        if (array.isEmpty()) {
            items.addAll(def);
        }
        return items;
    }

    public static void putTabList(Context context, String key, ArrayList<TabItem> tabs) {
        List<String> items = new ArrayList<>();
        for (TabItem item : tabs) {
            items.add(item.toString());
        }
        put(context).putString(key, TextUtils.join("‚‗‚", items)).apply();
    }

    public static void remove(Context context, String key) {
        put(context).remove(key).apply();
    }
}

File: manager/src/main/java/com/slim/slimfilemanager/settings/SettingsProvider.kt
/*
 * Copyright (C) 2015 The SlimRoms Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.slim.slimfilemanager.settings

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager
import android.text.TextUtils

import com.slim.slimfilemanager.widget.TabItem

import java.util.ArrayList
import java.util.Arrays

object SettingsProvider {

    // File Manager
    val KEY_ENABLE_ROOT = "enable_root"
    val THEME = "app_theme"
    val SORT_MODE = "sort_mode"
    val SMALL_INDICATOR = "small_indicator"

    // Text Editor
    val USE_MONOSPACE = "use_monospace"
    val EDITOR_WRAP_CONTENT = "editor_wrap_content"
    val SUGGESTION_ACTIVE = "suggestion_active"
    val EDITOR_ENCODING = "editor_encoding"
    val FONT_SIZE = "font_size"
    val SPLIT_TEXT = "page_system_active"

    operator fun get(context: Context): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(context)
    }

    fun put(context: Context): SharedPreferences.Editor {
        return get(context).edit()
    }

    fun put(context: Context, key: String, o: Any) {
        if (o is Boolean) {
            put(context).putBoolean(key, o).apply()
        } else if (o is String) {
            put(context).putString(key, o).apply()
        } else if (o is Int) {
            put(context).putInt(key, o).apply()
        }
    }

    fun getString(context: Context, key: String, defValue: String): String? {
        return get(context).getString(key, defValue)
    }

    fun putString(context: Context, key: String, value: String) {
        put(context).putString(key, value).apply()
    }

    fun getBoolean(context: Context, key: String, defValue: Boolean): Boolean {
        return get(context).getBoolean(key, defValue)
    }

    fun putBoolean(context: Context, key: String, b: Boolean) {
        put(context).putBoolean(key, b).apply()
    }

    fun getInt(context: Context, key: String, defValue: Int): Int {
        var i: Int
        try {
            i = get(context).getInt(key, defValue)
        } catch (e: Exception) {
            i = Integer.parseInt(get(context).getString(key, Integer.toString(defValue))!!)
        }

        return i
    }

    fun putInt(context: Context, key: String, value: Int) {
        put(context).putInt(key, value).commit()
    }

    fun getTabList(context: Context, key: String,
                   def: ArrayList<TabItem>): ArrayList<TabItem> {
        val items = ArrayList<TabItem>()
        val array = ArrayList(
                Arrays.asList(*TextUtils.split(get(context).getString(key, ""), "‚‗‚")))
        for (s in array) {
            items.add(TabItem.fromString(s))
        }
        if (array.isEmpty()) {
            items.addAll(def)
        }
        return items
    }

    fun putTabList(context: Context, key: String, tabs: ArrayList<TabItem>) {
        val items = ArrayList<String>()
        for (item in tabs) {
            items.add(item.toString())
        }
        put(context).putString(key, TextUtils.join("‚‗‚", items)).apply()
    }

    fun remove(context: Context, key: String) {
        put(context).remove(key).apply()
    }
}